src/main/java/net/minecraftforge/registries/NamespacedHolderHelper.java

```diff
@@ -120,6 +133,11 @@ boolean isKnownTagName(TagKey<T> name)
         return named;
     }
 
+    void addOptionalTag(TagKey<T> name, Set<Supplier<T>> defaults)
+    {
+        this.optionalTags.computeIfAbsent(name, k -> new HashSet<>()).addAll(defaults);
```

Comment:
I don't think this and the injecting of works quite right in terms of duplicates. As it is a set of suppliers it won't remove duplicates if there are two different instances supplying the same value. I think it may need a distinct call or something before the toList.